sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuShuffleExchangeExecBase.scala

```diff
@@ -312,6 +313,32 @@ object GpuShuffleExchangeExecBase {
   val METRIC_SHUFFLE_STALLED_BY_INPUT_STREAM = "rapidsShuffleStalledByInputStream"
   val METRIC_DESC_SHUFFLE_STALLED_BY_INPUT_STREAM =
     "RAPIDS shuffle time stalled by input stream operations"
+  
+  // ThreadLocal variable to track if current task has only one ColumnarBatch
+  private val singleBatchTaskLocal: ThreadLocal[Boolean] = new ThreadLocal[Boolean] {
```

Comment:
Lets not keep thread locals for this state.

Can we at least use `GpuColumnVector.tagAsFinalBatch`? I know `tagAsFinalBatch` is right now only applied in the sort and the coalesce. If the shuffle writer sees the first batch, and it's also final, then we know it's a single-batch task.

Reply:
ok, any particular reason why thread local is not favored?

Reply:
I added a ColumnVectorWithState, because tagAsFinalBatch cannot be used directly in our case